fdbserver/team/TestBase.actor.cpp

```diff
@@ -0,0 +1,749 @@
+	}
+
+	persistCommitRequest(pFakeTLog, teamID, items);
+
+	std::cout << "Current items in TLog: " << pFakeTLog->mutations.size() << std::endl;
+	std::cout << "<<< End TLog" << std::endl;
+
+	pFakeTLog->version = commitRequest.version;
+	commitRequest.reply.send(pFakeTLog->version);
+}
+
+ACTOR Future<Void> fakeTLog_ActivelyPush(std::shared_ptr<FakeTLog> pFakeTLog) {
+	state std::shared_ptr<TLogInterface_ActivelyPush> interface =
+	    std::dynamic_pointer_cast<TLogInterface_ActivelyPush>(pFakeTLog->interface);
+	if (interface == nullptr) {
+		// FIXME more proper exception type
+		throw internal_error();
+	}
+
+	loop {
+		choose {
+			when(TLogCommitRequest commitRequest = waitNext(interface->commit.getFuture())) {
+				std::cout << "Received commit request" << std::endl;
+				state TeamID teamID;
+				state uint32_t numMessages;
+				state std::vector<SubsequenceMutationItem> items;
+				processTLogCommitRequest(pFakeTLog, commitRequest, teamID, numMessages, items);
+				Future<Void> _ = pushCommitToStorageServer(pFakeTLog, teamID, items);
+			}
+		} // choose
+	} // loop
+}
+
+ACTOR Future<Void> fakeTLog_PassivelyProvide(std::shared_ptr<FakeTLog> pFakeTLog) {
+	state std::shared_ptr<TLogInterface_PassivelyPull> interface =
+	    std::dynamic_pointer_cast<TLogInterface_PassivelyPull>(pFakeTLog->interface);
+	if (interface == nullptr) {
+		// FIXME more proper exception type
+		throw internal_error();
+	}
+
+	loop {
+		choose {
+			when(TLogCommitRequest commitRequest = waitNext(interface->commit.getFuture())) {
+				TeamID teamID;
+				uint32_t numMessages;
+				std::vector<SubsequenceMutationItem> items;
+				processTLogCommitRequest(pFakeTLog, commitRequest, teamID, numMessages, items);
+			}
+			when(TLogPullRequest pullRequest = waitNext(interface->pullRequests.getFuture())) {
+
+			}
+		} // choose
+	} // loop
+}
+
+struct FakeStorageServer {
+	UID id = deterministicRandom()->randomUniqueID();
+
+	std::shared_ptr<StorageServerInterfaceBase> interface;
+};
+
+ACTOR Future<Void> fakeStorageServer_ActivePull(std::shared_ptr<FakeStorageServer> pFakeStorageServer) {
+	wait(delay(1.0));
+	return Void();
+}
+
+ACTOR Future<Void> fakeStorageServer_PassiveReceive(std::shared_ptr<FakeStorageServer> pFakeStorageServer) {
+	state std::shared_ptr<StorageServerInterface_PassivelyReceive> interface =
+	    std::dynamic_pointer_cast<StorageServerInterface_PassivelyReceive>(pFakeStorageServer->interface);
+	if (interface == nullptr) {
+		// FIXME more proper exception type
+		throw internal_error();
+	}
+
+	loop {
+		choose {
+			when(StorageServerPushRequest pushRequest = waitNext(interface->pushRequests.getFuture())) {
+				std::cout << ">>> Begin StorageServer" << std::endl;
+				std::cout << "TeamID: " << pushRequest.teamID.toString() << std::endl;
+				std::cout << "Version: " << pushRequest.version << std::endl;
+				std::cout << "Number of mutations: " << pushRequest.mutations.size() << std::endl;
```

Comment:
Instead of manual check, you can save the original messages in memory and check here the received messages are correct.